core-metastore/src/main/java/org/zalando/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -241,4 +292,235 @@ public String toString() {
                     '}';
         }
     }
+
+    private ListResult listBackwardToken(
+            final StringBuilder queryBuilder, final List<String> clauses, final List<Object> params,
+            final Token token, final int limit) {
+
+        clauses.add(" s_id " + token.getTokenType().dbClause + " ? ");
+        params.add(token.getSubscriptionId());
+
+        final List<Subscription> subscriptionsPlusOne = executeListSubscriptions(
+                queryBuilder, clauses, params, limit + 1, false);
+        // subscriptions are listed in a backwards direction, therefore they should be reversed afterwards
+        Collections.reverse(subscriptionsPlusOne);
+        if (subscriptionsPlusOne.size() > limit) {
+            // we need to remove the first one, as it shows that there is something before.
+            return new ListResult(
+                    subscriptionsPlusOne.subList(1, subscriptionsPlusOne.size()),
+                    new Token(subscriptionsPlusOne.get(limit).getId(), TokenType.FORWARD),
+                    new Token(subscriptionsPlusOne.get(0).getId(), TokenType.BACKWARD)
```

Comment:
I think here is a bug. It should be either `BACKWARD_INCL` or `subscriptionsPlusOne.get(1)`

Reply:
I need to double-check, but I didn't change this line.

Reply:
I find the logic here to be over-complicated.  Is anyone using the backward listing at all?  We could greatly simplify it if we removed this possibility.,